sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -155,6 +172,25 @@ public boolean start() throws IOException {
     return advance();
   }
 
+  private SchemaRegistryClient getSchemaRegistryClient(Read<K, V> spec) {
+    SchemaRegistryClient registryClient;
+    String schemaRegistryURL =
+        (String)
+            spec.getConsumerConfig().get(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG);
```

Comment:
Can you make `AbstractKafkaAvroSerDeConfig` into `KafkaAvroDeserializerConfig` as below.